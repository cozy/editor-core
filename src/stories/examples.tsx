import React from 'react';

import { boolean, object, radios, withKnobs } from '@storybook/addon-knobs';
import { IntlProvider } from 'react-intl-next';

import Editor, { EditorProps } from '../editor';
import EditorContext from '../ui/EditorContext';

function WrapperEditorComponent(props: EditorProps) {
  const ChildKey = JSON.stringify(props);
  return (
    <EditorContext>
      <IntlProvider locale="en">
        <Editor key={ChildKey} {...props} />
      </IntlProvider>
    </EditorContext>
  );
}

export default {
  title: 'Editor',
  component: Editor,
  decorators: [withKnobs],
};

const groupId = 'GROUP-ID1';
const Template = (args: any) => (
  <WrapperEditorComponent
    allowUndoRedoButtons={boolean('Undo/Redo Buttons', false, groupId)}
    allowPanel={boolean('Panel', false, groupId)}
    appearance={radios(
      'Appearance',
      {
        Comment: 'comment',
        'Full Page': 'full-page',
      },
      'full-page',
      groupId,
    )}
    allowTables={object('Table Options', {}, groupId)}
    {...args}
  />
);

export const EditorExample = Template.bind({});
