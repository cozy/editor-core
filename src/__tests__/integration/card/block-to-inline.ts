import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { ConfluenceCardProvider } from '../../../../examples/5-full-page-with-confluence-smart-cards';
import * as blockCardAdf from './_fixtures_/block-card-with-empty-p-at-bottom.adf.json';
import { waitForBlockCardSelection } from '@atlaskit/media-integration-test-helpers';
import { getDocFromElement, editable, getProseMirrorPos } from '../_helpers';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';

BrowserTestCase(
  `card: should switch to inline mode with proper cursor position`,
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    const cardProviderPromise = Promise.resolve(
      new ConfluenceCardProvider('prod'),
    );

    await mountEditor(page, {
      appearance: 'full-page',
      allowTextAlignment: true,
      defaultValue: JSON.stringify(blockCardAdf),
      UNSAFE_cards: {
        provider: cardProviderPromise,
        allowBlockCards: true,
      },
    });

    await waitForBlockCardSelection(page);
    await page.click('div[aria-label="Floating Toolbar"] button');
    // Click the "Display inline" to switch to inline card
    await page.click('[data-testid="inline-appearance"]');

    expect(await getProseMirrorPos(page)).toEqual(4);
    expect(
      await page.$eval(editable, getDocFromElement),
    ).toMatchCustomDocSnapshot(testName);
  },
);
