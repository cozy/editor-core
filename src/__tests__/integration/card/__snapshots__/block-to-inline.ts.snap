// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card: should switch to inline mode with proper cursor position 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "data": Object {
              "@context": "https://www.w3.org/ns/activitystreams",
              "@type": "Document",
              "generator": Object {
                "icon": "https://wac-cdn.atlassian.com/assets/img/favicons/atlassian/favicon.png",
              },
              "name": "Welcome to Atlassian!",
              "url": "http://www.atlassian.com",
            },
            "url": null,
          },
          "type": "inlineCard",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
